### ISSUE TITLE
[P110] Handshake failing in latest firmware

### ISSUE THREAD

**Issue by achetronic** [richardjennings/tapo#4]
Hello there. First, congratulations for this amazing project. It's being used as library inside https://github.com/achetronic/autoheater/

With the latest version of the firmware on P110 plugs (updated not in purpose, but forgot disable automatic updates) I got the following message:

```console
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
github.com/richardjennings/tapo/pkg/tapo.(*Tapo).Handshake(0xc000146000)
	/home/achetronic/go/go1.17.1/pkg/mod/github.com/richardjennings/tapo@v0.0.0-20230101130552-aebccaf799d8/pkg/tapo/tapo.go:139 +0x8f0
github.com/richardjennings/tapo/pkg/tapo.NewTapo({0x7fff14f9d76f, 0x300000002}, {0x7fff14f9d77d, 0x13}, {0x7fff14f9d791, 0xe})
	/home/achetronic/go/go1.17.1/pkg/mod/github.com/richardjennings/tapo@v0.0.0-20230101130552-aebccaf799d8/pkg/tapo/tapo.go:55 +0x2ed
main.main()
	/home/achetronic/go/go1.17.1/pkg/mod/github.com/richardjennings/tapo@v0.0.0-20230101130552-aebccaf799d8/main.go:25 +0xd6
```

I assume this is caused by a change on Tapo API spec. Do you know something about? :)

**Comment by arigon** [comment]
Tapo changed the protocol to KLAP. See here: https://github.com/evcc-io/evcc/discussions/10461

**Comment by achetronic** [comment]
@arigon @tazerreloaded Thank you for guiding me to the problem. I supposed it but needed the exact names

The library maintainer does not seem to be maintaining this anymore, so finally I did some research, tests and ports and got it working https://github.com/achetronic/tapogo

Mine is only supporting the latest KLAP protocol. I arranged it in a reusable way just in case the maintainer of this one wants to introduce KLAP on this. Hope my research will be useful for everyone 🙂
